framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -2292,4 +2304,23 @@ protected Properties getProperties(Class<?> cls, String filePath) {
     public final SourceVersion getSupportedSourceVersion() {
         return SourceVersion.latest();
     }
+
+    /** Print information about the git repository from which the Checker Framework was compiled. */
+    void printGitProperties() {
+        if (gitPropertiesPrinted) {
+            return;
+        }
+        gitPropertiesPrinted = true;
+
+        InputStream in = getClass().getResourceAsStream("/git.properties");
```

Comment:
Wouldn't it be better to use try-with-resources for this, to make sure the files are closed in all cases?